haystack/preprocessor/preprocessor.py

```diff
@@ -76,9 +76,48 @@ def process(
         split_overlap: Optional[int] = None,
         split_respect_sentence_boundary: Optional[bool] = None,
     ) -> List[dict]:
+
         """
-        Perform document cleaning and splitting. Takes a single document as input and returns a list of documents.
+        Perform document cleaning and splitting. Can takes a single document or a list of documents as input and returns a list of documents.
         """
+
+        kwargs = {
+            "clean_whitespace": clean_whitespace,
+            "clean_header_footer": clean_header_footer,
+            "clean_empty_lines": clean_empty_lines,
+            "split_by": split_by,
+            "split_length": split_length,
+            "split_overlap": split_overlap,
+            "split_respect_sentence_boundary": split_respect_sentence_boundary
+        }
+
+        ret = []
+
+        if type(documents) == dict:
+            ret = self._process_single(
+                document=documents,
+                **kwargs                #type: ignore
+        )
+        elif type(documents) == list:
+            ret = self._process_batch(
+                documents=list(documents),
+                **kwargs
+            )
+
```

Comment:
Let's raise an error if it's neither list nor dict. Right now we just return an empty list